essentials-framework/Essentials Core/PepperDashEssentialsBase/Comm and IR/CecPortController.cs

```diff
@@ -57,8 +61,12 @@ void OnDataReceived(string s)
                 bytesHandler(this, new GenericCommMethodReceiveBytesArgs(bytes));
             }
             var textHandler = TextReceived;
-            if (textHandler != null)
-                textHandler(this, new GenericCommMethodReceiveTextArgs(s));
+			if (textHandler != null)
+			{
+				if (StreamDebugging.RxStreamDebuggingIsEnabled)
```

Comment:
This check and debug statement should be moved to the top of this method. This way, the string will be printed when stream debugging is enabled even if there are no subscribers to the methods.

Reply:
After review with Evan, we kept the `if(StreamDebugging...` in the event check, and added a statement in the bytesHandler check as well. Marking as resolved.